File: src/main/java/entities/Parametres2FA.java
package entities;

import java.time.LocalDateTime;

/**
 * Entité Parametres2FA
 * Représente les paramètres de sécurité 2FA d'un utilisateur
 */
public class Parametres2FA {

    private int id;
    private int utilisateurId;
    private boolean estActive;
    private String methodePreferee; // email, sms, desactive
    private String telephone2fa;
    private LocalDateTime dateActivation;

    // Constructeurs
    public Parametres2FA() {
        this.estActive = false;
        this.methodePreferee = "email";
    }

    public Parametres2FA(int utilisateurId) {
        this.utilisateurId = utilisateurId;
        this.estActive = false;
        this.methodePreferee = "email";
    }

    // Getters et Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getUtilisateurId() {
        return utilisateurId;
    }

    public void setUtilisateurId(int utilisateurId) {
        this.utilisateurId = utilisateurId;
    }

    public boolean isEstActive() {
        return estActive;
    }

    public void setEstActive(boolean estActive) {
        this.estActive = estActive;
    }

    public String getMethodePreferee() {
        return methodePreferee;
    }

    public void setMethodePreferee(String methodePreferee) {
        this.methodePreferee = methodePreferee;
    }

    public String getTelephone2fa() {
        return telephone2fa;
    }

    public void setTelephone2fa(String telephone2fa) {
        this.telephone2fa = telephone2fa;
    }

    public LocalDateTime getDateActivation() {
        return dateActivation;
    }

    public void setDateActivation(LocalDateTime dateActivation) {
        this.dateActivation = dateActivation;
    }

    @Override
    public String toString() {
        return "Parametres2FA{" +
                "id=" + id +
                ", utilisateurId=" + utilisateurId +
                ", estActive=" + estActive +
                ", methodePreferee='" + methodePreferee + '\'' +
                '}';
    }
}

File: src/Entity/Parametres2FA.php
<?php

namespace App\Entity;

use App\Repository\Parametres2FARepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: Parametres2FARepository::class)]
class Parametres2FA
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne]
    private ?Utilisateur $utilisateur_id = null;

    #[ORM\Column(nullable: true)]
    private ?bool $est_active = null;

    #[ORM\Column(length: 100, nullable: true)]
    private ?string $methode_preferee = null;

    #[ORM\Column(length: 20, nullable: true)]
    private ?string $telephone_2fa = null;

    #[ORM\Column(nullable: true)]
    private ?\DateTime $date_activation = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function setId(int $id): static
    {
        $this->id = $id;

        return $this;
    }

    public function getUtilisateurId(): ?Utilisateur
    {
        return $this->utilisateur_id;
    }

    public function setUtilisateurId(?Utilisateur $utilisateur_id): static
    {
        $this->utilisateur_id = $utilisateur_id;

        return $this;
    }

    public function isEstActive(): ?bool
    {
        return $this->est_active;
    }

    public function setEstActive(?bool $est_active): static
    {
        $this->est_active = $est_active;

        return $this;
    }

    public function getMethodePreferee(): ?string
    {
        return $this->methode_preferee;
    }

    public function setMethodePreferee(?string $methode_preferee): static
    {
        $this->methode_preferee = $methode_preferee;

        return $this;
    }

    public function getTelephone2fa(): ?string
    {
        return $this->telephone_2fa;
    }

    public function setTelephone2fa(?string $telephone_2fa): static
    {
        $this->telephone_2fa = $telephone_2fa;

        return $this;
    }

    public function getDateActivation(): ?\DateTime
    {
        return $this->date_activation;
    }

    public function setDateActivation(?\DateTime $date_activation): static
    {
        $this->date_activation = $date_activation;

        return $this;
    }
}
